packages/devtools_app/lib/src/memory/memory_screen.dart

```diff
@@ -345,10 +367,188 @@ class MemoryBodyState extends State<MemoryBody> with AutoDisposeMixin {
             includeTextWidth: _primaryControlsMinVerboseWidth,
           ),
         ),
+        const SizedBox(width: defaultSpacing),
+        OutlineButton(
+          key: legendKey,
+          onPressed: controller.toggleLegendVisibility,
+          child: MaterialIconLabel(
+            legendOverlayEntry == null ? Icons.storage : Icons.close,
+            'Legend',
+            includeTextWidth: _primaryControlsMinVerboseWidth,
+          ),
+        ),
       ],
     );
   }
 
+  final legendKey = GlobalKey(debugLabel: MemoryScreen.legendKeyName);
+  static const legendXOffset = 20;
+  static const legendYOffset = 7.0;
+  static const legendWidth = 200.0;
+  static const legendTextWidth = 55.0;
+  static const legendHeight1Chart = 185.0;
+  static const legendHeight2Charts = 340.0;
+
+  // TODO(terry): Consider custom painter?
```

Comment:
A custom painter would be nice so that these stay in sync with the images used in the chart itself.